### PR TITLE
x11-terms/gnome-terminal: bump to 3.46.1, x11-libs/vte: bump to 0.70.0, gui-libs/vte: new package 0.70.0 (gtk4 version) gui-libs/vte-common: new package ver 0.70.0

### PR DESCRIPTION
**Pull request by Schievel1**
@mattst88 , I made the patch (cntr-ntfy-autottl-ts) work for this version. I included it in the files directory. Please upload it to your g.o webspace. I prepared the ebuild so you just have to uncomment/ delete lines once its uploaded 

This PR bumps x11-terms/gnome-terminal. Since x11-libs/vte is a dependency, it also bumps that. 
x11-libs/vte has the option to be built with GTK3 and GTK4 support. Because of https://wiki.gentoo.org/wiki/Project:GNOME/Gnome_Team_Ebuild_Policies#GTK.2B_USE-flag_and_slots_usage there should not be a x11-libs/vte with a GTK4-USE-flag. Instead, this should be a new slot. 
Since we are moving away from the x11-* directories, I made that a new package in gui-libs/ instead. (See also x11-libs/gtk+ and gui-libs/gtk)
Because x11-libs/vte and gui-libs/vte have package collisons, I stripped both packages from the collisions and put them into gui-libs/vte-common, which both, gui-libs/vte and x11-libs/vte depend on.

### CODE REVIEW

File: gui-libs/vte-common/vte-common-0.70.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+PYTHON_COMPAT=( python3_{8..10} )
+
+inherit gnome.org meson python-any-r1
+
+DESCRIPTION="Library providing a virtual terminal emulator widget"
+HOMEPAGE="https://wiki.gnome.org/Apps/Terminal/VTE https://gitlab.gnome.org/GNOME/vte"
+
+# Once SIXEL support ships (0.66 or later), might need xterm license (but code might be considered upgraded to LGPL-3+)
+LICENSE="LGPL-3+ GPL-3+"
+SLOT="2.91" # vte_api_name in meson.build
```
Comment (mattst88) [inline gui-libs/vte-common/vte-common-0.70.0.ebuild:14]: I think you mean `vte_api_version`?

File: gui-libs/vte/vte-0.70.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..10} )
+VALA_MIN_API_VERSION="0.48"
```
Comment (mattst88) [inline gui-libs/vte/vte-0.70.0.ebuild:6]: Can drop this.

File: x11-libs/vte/vte-0.70.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..10} )
+VALA_MIN_API_VERSION="0.48"
```
Comment (mattst88) [inline x11-libs/vte/vte-0.70.0.ebuild:6]: Can drop this.

File: x11-libs/vte/vte-0.70.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,100 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..10} )
```
Comment (mattst88) [inline x11-libs/vte/vte-0.70.0.ebuild:5]: Add Python 3.11.

File: gui-libs/vte-common/vte-common-0.70.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,78 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..10} )
```
Comment (mattst88) [inline gui-libs/vte-common/vte-common-0.70.0.ebuild:5]: Add Python 3.11.

File: gui-libs/vte/vte-0.70.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+PYTHON_COMPAT=( python3_{8..10} )
```
Comment (mattst88) [inline gui-libs/vte/vte-0.70.0.ebuild:5]: Add Python 3.11.

File: x11-terms/gnome-terminal/gnome-terminal-3.46.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+SRC_URI="https://gitlab.gnome.org/GNOME/${PN}/-/archive/${PV}/${P}.tar.gz"
+# SRC_URI+=" !vanilla? ( https://dev.gentoo.org/~mattst88/distfiles/${PN}-3.46.1-cntr-ntfy-autottl-ts.patch.xz )" # uncomment when patch is hosted
+
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~ia64 ~mips ~ppc ~ppc64 ~riscv ~sparc ~x86 ~amd64-linux ~x86-linux"
+
+# FIXME: automagic dependency on gtk+[X], just transitive but needs proper control, bug 624960
+RDEPEND="
+	>=dev-libs/glib-2.52:2
+	>=x11-libs/gtk+-3.22.27:3
+	>=x11-libs/vte-0.70.0:291[!vanilla?]
```
Comment (mattst88) [inline x11-terms/gnome-terminal/gnome-terminal-3.46.1.ebuild:23]: `:291` should be `:2.91`.

File: gui-libs/vte/vte-0.70.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+	introspection? ( >=dev-libs/gobject-introspection-1.56:= )
+	x11-libs/pango[introspection?]
+	=gui-libs/vte-common-${PV}[systemd?]
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/libxml2:2
+	dev-util/glib-utils
+	gtk-doc? ( dev-util/gi-docgen )
```
Comment (mattst88) [inline gui-libs/vte/vte-0.70.0.ebuild:43]: We need to add REQUIRED_USE="gtk-doc? ( introspection )"` whenever we begin using gi-docgen.

File: x11-libs/vte/vte-0.70.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+	introspection? ( >=dev-libs/gobject-introspection-1.56:= )
+	x11-libs/pango[introspection?]
+	=gui-libs/vte-common-${PV}[systemd?]
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	${PYTHON_DEPS}
+	dev-libs/libxml2:2
+	dev-util/glib-utils
+	gtk-doc? ( dev-util/gi-docgen )
```
Comment (mattst88) [inline x11-libs/vte/vte-0.70.0.ebuild:43]: We need to add REQUIRED_USE="gtk-doc? ( introspection )"` whenever we begin using gi-docgen.

File: gui-libs/vte/vte-0.70.0.ebuild [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+		-Dgtk4=true
+		$(meson_use icu)
+		$(meson_use systemd _systemd)
+		$(meson_use vala vapi)
+	)
+	meson_src_configure
+}
+
+src_install() {
+	meson_install # not meson_src_install because this would include einstalldocs, which would result in file collissions with x11-libs/vte
```
Comment (mattst88) [inline gui-libs/vte/vte-0.70.0.ebuild:87]: Let's fix the typo in `collisions` while we're here.

File: gui-libs/vte/vte-0.70.0.ebuild [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+#   # remove the files that have collissions with x11-libs/vte. The files come with gui-libs/vte-common instead
```
Comment (mattst88) [inline gui-libs/vte/vte-0.70.0.ebuild:88]: Let's fix the typo in `collisions` while we're here.

Also, let's remove the stray `#` at the beginning of the line to make the comment properly indented.

File: x11-libs/vte/vte-0.70.0.ebuild [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+		$(meson_use icu)
+		$(meson_use systemd _systemd)
+		$(meson_use vala vapi)
+	)
+	meson_src_configure
+}
+
+src_install() {
+	meson_install # not meson_src_install because this would include einstalldocs, which would result in file collissions with gui-libs/vte
+#   # remove the files that have collissions with gui-libs/vte. The files come with gui-libs/vte-common instead
```
Comment (mattst88) [inline x11-libs/vte/vte-0.70.0.ebuild:88]: Let's fix the typo in `collisions` while we're here.

Also, let's remove the stray `#` at the beginning of the line to make the comment properly indented.

File: x11-libs/vte/vte-0.70.0.ebuild [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+		-Dgtk4=false
+		$(meson_use icu)
+		$(meson_use systemd _systemd)
+		$(meson_use vala vapi)
+	)
+	meson_src_configure
+}
+
+src_install() {
+	meson_install # not meson_src_install because this would include einstalldocs, which would result in file collissions with gui-libs/vte
```
Comment (mattst88) [inline x11-libs/vte/vte-0.70.0.ebuild:87]: Let's fix the typo in `collisions` while we're here.

File: gui-libs/vte-common/vte-common-0.70.0.ebuild [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+		-Dgtk4=false
+		-Dicu=false
+		$(meson_use systemd _systemd)
+		-Dvapi=false
+	)
+	meson_src_configure
+}
+
+src_install() {
+	exeinto "/usr/libexec/"
```
Comment (mattst88) [inline gui-libs/vte-common/vte-common-0.70.0.ebuild:68]: I'd drop the quotes where they aren't needed (like here).

File: gui-libs/vte-common/vte-common-0.70.0.ebuild [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+	doexe "${BUILD_DIR}/src/vte-urlencode-cwd"
+	insinto "/etc/profile.d/"
```
Comment (mattst88) [inline gui-libs/vte-common/vte-common-0.70.0.ebuild:70]: and here.

File: gui-libs/vte-common/vte-common-0.70.0.ebuild [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+	newins "${BUILD_DIR}/src/vte.sh" "vte-${SLOT}.sh"
+	newins "${BUILD_DIR}/src/vte.csh" "vte-${SLOT}.csh"
+	if  use systemd; then
+	  insinto "/usr/lib/systemd/user/vte-spawn-.scode.d/"
```
Comment (mattst88) [inline gui-libs/vte-common/vte-common-0.70.0.ebuild:74]: and here, etc.

**Review by mattst88 — CHANGES_REQUESTED** [review]

**Review by mattst88 — APPROVED** [review]
Looks good. Thank you. I'd separate the `vte-common` stuff out of the commit that adds `gui-libs/vte` (I'll do that). It's best to have one commit per package.

File: gui-libs/vte/vte-0.70.0.ebuild [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+	>=dev-libs/glib-2.52:2
+	crypt?  ( >=net-libs/gnutls-3.2.7:0= )
+	icu? ( dev-libs/icu:= )
+	>=x11-libs/pango-1.22.0
+	>=dev-libs/libpcre2-10.21
+	systemd? ( >=sys-apps/systemd-220:= )
+	sys-libs/zlib
+	introspection? ( >=dev-libs/gobject-introspection-1.56:= )
+	x11-libs/pango[introspection?]
+	=gui-libs/vte-common-${PV}[systemd?]
```
[the remark this answers is not in the document]
Comment (Schievel1) [reply gui-libs/vte/vte-0.70.0.ebuild:38]: noted, thanks for your input

**Pull request closed without merging**